atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java

```diff
@@ -110,60 +114,79 @@ public void run() {
             Thread.sleep(getBackoffTimeWhenSweepHasNotRun());
             log.info("Starting background sweeper.");
             while (true) {
-                long millisToSleep = checkConfigAndRunSweep(locks);
-                Thread.sleep(millisToSleep);
+                SweepOutcome outcome = checkConfigAndRunSweep(locks);
+
+                updateBatchSize(outcome);
+                updateMetrics(outcome);
+
+                sleepUntilNextRun(outcome);
             }
         } catch (InterruptedException e) {
             log.warn("Shutting down background sweeper. Please restart the service to rerun background sweep.");
         }
     }
 
-    // Returns milliseconds to sleep
+    private void updateBatchSize(SweepOutcome outcome) {
+        if (outcome == SweepOutcome.SUCCESS) {
+            batchSizeMultiplier = Math.min(1.0, batchSizeMultiplier * 1.01);
+            return;
+        }
+        if (outcome == SweepOutcome.RETRYING_WITH_SMALLER_BATCH) {
+            SweepBatchConfig lastBatchConfig = specificTableSweeper.getAdjustedBatchConfig();
+
+            // Cut batch size in half, always sweep at least one row (we round down).
+            batchSizeMultiplier = Math.max(batchSizeMultiplier / 2, 1.5 / lastBatchConfig.candidateBatchSize());
```

Comment:
We would change the multiplier after the first time in the past but I think it makes more sense to do it before, just noting that this behavior has changed.

Reply:
Actually we should change it later, otherwise `specificTableSweeper.getSweepBatchConfig()` will not return the previous sweep batch configs for logging.

Reply:
All of the logging that needs the old config either happens before here or (in the line below) is using the already computed `lastBatchConfig `.

Is there some other behaviour I've accidentally changed here?